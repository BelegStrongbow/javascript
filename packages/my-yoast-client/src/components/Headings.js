import styled from "styled-components";
import colors from "yoast-components/style-guide/colors.json";
import defaults from "../config/defaults.json";

export const ListHeading = styled.span`
	display: block;
	font-size: 1em;
	padding: 24px 40px;
	font-weight: 400;
	margin: 0;
	border-bottom: 1px solid ${ colors.$color_grey_medium };

	@media screen and ( max-width: ${ defaults.css.breakpoint.tablet }px ) {
		padding: 24px;
	}

	@media screen and ( max-width: ${ defaults.css.breakpoint.mobile }px ) {
		padding: 16px;
	}
`;

export const Heading = styled.h1`
	font-weight: 700;
	font-size: 2em;
`;

export const ModalHeading = styled.h1`
	font-weight: 300;
	font-size: 1.5em;
	margin: 0;
`;

export const SubHeading = styled.h2`
	font-weight: 300;
	font-size: 1.3em;
	margin: 0;
`;
