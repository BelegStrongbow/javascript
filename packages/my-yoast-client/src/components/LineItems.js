import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";
import defaults from "../config/defaults.json";

const ItemsContainer = styled.ul`
	padding: 0;
	list-style: none;
	flex-shrink: 0;

	@media screen and ( max-width: ${ defaults.css.breakpoint.mobile }px ) {
		overflow: hidden;
	}
`;

/**
 * LineItems.
 *
 * @param {Object} props Properties to be passed to the table.
 * @returns {ReactElement} A row of LineItems.
 */
function LineItems( props ) {
	const items = props.items.map( ( item ) => {
		return <li key={ item.id }>{ item.quantity }<span aria-hidden="true">&times;</span> { item.productName }</li>;
	} );
	return <ItemsContainer>{ items }</ItemsContainer>;
}

LineItems.propTypes = {
	items: PropTypes.array,
};

LineItems.defaultProps = {
	items: [],
};

export default LineItems;
