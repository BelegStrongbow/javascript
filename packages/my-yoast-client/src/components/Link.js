import PropTypes from "prop-types";
import React, { Component } from "react";
import { Link as RouterLink } from "react-router-dom";

import NewTabMessage from "./NewTabMessage";

/**
 * Determines if a certain URL is an external URL.
 *
 * @param {string} url The URL to test.
 * @returns {boolean} Whether or not the URL is an external URL.
 */
function isExternal( url ) {
	return url.startsWith( "http" ) || url === "#" || url.startsWith( "mailto:" );
}

/**
 * Determines if a certain URL points to Yoast.
 *
 * @param {string} url The URL to test.
 * @returns {boolean} Whether or not the URL points to Yoast.
 */
function isYoastLink( url ) {
	return /yoast\.com|yoast\.test|yoa\.st/.test( url );
}

/**
 * Link that also works for external URL's, see https://github.com/ReactTraining/react-router/issues/1147
 * for more information.
 *
 * When an external link has a `target="_blank"` attribute, it will automatically
 * add the accessibility visually hidden message "(Opens in a new browser tab)"
 * by using the `NewTabMessage` component.
 *
 */
export default class Link extends Component {
	/**
	 * Renders the component.
	 *
	 * @returns {ReactElement} The rendered component.
	 */
	render() {
		const internalProps = Object.assign( {}, this.props );
		delete internalProps.linkTarget;
		delete internalProps.linkRel;
		delete internalProps.ariaLabel;
		delete internalProps.enabledStyle;
		delete internalProps.iconSource;
		delete internalProps.iconSize;

		/*
		 * When a link has a target _blank attribute and it doesn't point to Yoast,
		 * we always want a default "noopener" rel attribute value.
		 */
		let relValueForTargetBlank = "noopener";

		/*
		 * When a link has a target _blank attribute and it does point to Yoast,
		 * we use the linkRel prop which defaults to `null` so it doesn't render
		 * the rel attribute. This way, it can be overridden setting the prop, if necessary.
		 */
		if ( isYoastLink( this.props.to ) ) {
			relValueForTargetBlank = this.props.linkRel;
		}

		const hasTargetBlank = this.props.linkTarget === "_blank";

		const externalProps = {
			href: this.props.to,
			className: this.props.className,
			"aria-label": this.props.ariaLabel,
			target: this.props.linkTarget,
			rel: hasTargetBlank ? relValueForTargetBlank : this.props.linkRel,
			onClick: this.props.onClick,
		};

		return isExternal( this.props.to )
			? <a { ...externalProps }>
				{ this.props.children }
				{ hasTargetBlank && <NewTabMessage /> }
			</a>
			: <RouterLink { ...internalProps } />;
	}
}

Link.propTypes = {
	to: PropTypes.string.isRequired,
	className: PropTypes.string,
	children: PropTypes.any.isRequired,
	ariaLabel: PropTypes.string,
	linkTarget: PropTypes.string,
	linkRel: PropTypes.string,
	onClick: PropTypes.func,
};

Link.defaultProps = {
	className: null,
	ariaLabel: null,
	linkTarget: null,
	linkRel: null,
	onClick: null,
};
