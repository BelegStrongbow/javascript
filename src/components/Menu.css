.menuItemActive {
    display: block;
    padding-left: 0px;
    color: purple;
    box-shadow: inset 0px 2px 8px 0px rgba(0, 0, 0, 0.3);
    font-family: 'open sans 400';
}