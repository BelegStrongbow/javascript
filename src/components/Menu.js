import React from "react";
import styled from "styled-components";
import colors from "yoast-components/style-guide/colors.json";
import { NavLink, Route } from "react-router-dom";

const activeStyle = "active-class-name";

const Menu = styled.nav`
	clear: both;
	

	& ul {
		list-style: none;
		margin: 0;
		padding: 0;
	}

	& li {
		display: inline;
	}

	@media screen and ( max-width: 1024px ) {
	margin: 0 4%;
	text-align: center;
	
	
	& ul {
		display: flex;
	}
		
	& li {
		flex: 1 1 0%; /* Has unit measure to ensure IE compatibility */
		min-width: 40px; /* Like the icons width. */
	}
	}
`;

const MenuItem = styled( NavLink )`
	display: block;
	height: 100px;
	line-height: 100px;
	font-size: 22px;
	font-weight: 300;
	padding-left: 25px;
	margin-left: 25px;
	padding-right: 25px;
	color: ${colors.$color_background_light};
	text-decoration: none;

	&.${ activeStyle } {
		color: ${colors.$color_border};
		background-color: ${colors.$color_grey_light};
		box-shadow: inset 12px 0px 6px -10px rgba(0, 0, 0, 0.3);
		font-weight: 400;
		color: ${colors.$color_pink_dark};
		position: relative;
		overflow-y: hidden;

		&:before,
		&:after {
			content: "";
			display: block;
			width: 100%;
			height: 5px;
			position: absolute;
			left: 0;
			background: transparent;
		}

		&:before {
			border-bottom-right-radius: 20px 10px;
			top: -5px;
			box-shadow: 0px 2px 8px 0px rgba(0, 0, 0, 0.3);
		}

		&:after {
			border-top-right-radius: 20px 10px;
			bottom: -5px;
			box-shadow: 0px -1px 8px 0px rgba(0, 0, 0, 0.3);
		}
	}

	@media screen and ( max-width: 1024px ) {
		display: inline-block;
		width: 100%; /* necessary for the text ellipsis */
		
		height: 74px;
		margin: 0;
		padding: 8px 0 0;
		border-bottom: 5px solid transparent;
		box-shadow: none;
		color: ${colors.$color_white};
		font-size: 12px;
		font-weight: 400;
		line-height: inherit;
		overflow: hidden;
		white-space: nowrap;
		text-overflow: ellipsis;

		&.${ activeStyle } {
			border-bottom: 5px solid ${colors.$color_white};
			color: ${colors.$color_white};
			background-color: transparent;
			box-shadow: none;

			&:before,
			&:after {
				content: none;
			}
		}
	}
`;

const MenuIcon = styled.img`
	display: none;

	@media screen and ( max-width: 1024px ) {
		display: block;
		width: 40px;
		height: 40px;
		margin: 0 auto -3px;
	}
`;

/**
 * The main menu.
 *
 * @param {Object} props The props to use.
 * @returns {ReactElement} The rendered component.
 */
export function MainMenu( props ) {
	return (
	<Menu>
		<ul role="list">
			{ props.menuRoutes.map( function( page ) {
				let isActive = page.isActive || ( ( match ) => {
					return match;
				} );

				return <li key={ page.title }>
					<MenuItem activeClassName={ activeStyle } to={ page.path } isActive={ isActive }>
						<MenuIcon src={ page.icon } alt="" />
						{ page.title }
					</MenuItem>
				</li>;
			}
			) }
		</ul>
	</Menu>
	);
}

MainMenu.propTypes = {
	menuRoutes: React.PropTypes.array.isRequired,
};


/**
 * The main menu routes.
 *
 * @param {Object} props The props to use.
 * @returns {ReactElement} The rendered component.
 */
export function MainMenuRoutes( props ) {
	return (
	<div>
		{ props.menuRoutes.map( function( route, routeKey ) {
			return <Route key={ routeKey } path={ route.path } component={ route.component }/>;
		}
		) }
	</div>
	);
}

MainMenuRoutes.propTypes = {
	menuRoutes: React.PropTypes.array.isRequired,
};
