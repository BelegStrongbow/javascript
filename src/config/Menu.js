// External dependencies.
import React from "react";
import { defineMessages } from "react-intl";

// Internal dependencies.
import HomeIcon from "../icons/Home.js";
import SitesIcon from "../icons/Sites.js";
import CoursesIcon from "../icons/Courses.js";
import UserIcon from "../icons/User.js";
import DownloadsIcon from "../icons/Download.js";
import SitesPageContainer from "../containers/SitesPage";
import AccountPage from "../components/AccountPage";
import DownloadsPage from "../containers/DownloadsPage";
import CoursesPage from "../containers/CoursesPage";
import HomePage from "../containers/HomePage";

const titles = defineMessages( {
	home: {
		id: "menu.title.home",
		defaultMessage: "Home",
	},
	sites: {
		id: "menu.title.sites",
		defaultMessage: "Sites",
	},
	courses: {
		id: "menu.title.courses",
		defaultMessage: "Courses",
	},
	downloads: {
		id: "menu.title.downloads",
		defaultMessage: "Downloads",
	},
	account: {
		id: "menu.title.account",
		defaultMessage: "Account",
	},
} );

const menuItems = [
	{
		showInMenu: true,
		path: "/",
		titleKey: "home",
		iconSource: <HomeIcon />,
		component: HomePage,
		exact: true,
		isActive: ( match, location ) => {
			return location.pathname === "/";
		},
		title: titles.home,
	},
	{
		showInMenu: true,
		path: "/sites",
		titleKey: "sites",
		iconSource: <SitesIcon />,
		component: SitesPageContainer,
		exact: true,
		title: titles.sites,
	},
	{
		showInMenu: true,
		path: "/courses",
		titleKey: "courses",
		iconSource: <CoursesIcon />,
		component: CoursesPage,
		exact: false,
		title: titles.courses,
	},
	{
		showInMenu: true,
		path: "/downloads",
		titleKey: "downloads",
		iconSource: <DownloadsIcon />,
		component: DownloadsPage,
		exact: true,
		title: titles.downloads,
	},
	{
		showInMenu: true,
		path: "/account",
		titleKey: "account",
		iconSource: <UserIcon />,
		component: AccountPage,
		exact: false,
		title: titles.account,
	},
];

export default menuItems;
