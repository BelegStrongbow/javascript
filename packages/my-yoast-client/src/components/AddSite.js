import PropTypes from "prop-types";
import React from "react";
import { injectIntl, intlShape, defineMessages, FormattedMessage } from "react-intl";
import { LargeButton, makeButtonFullWidth, LargeSecondaryButton } from "./Button.js";
import styled from "styled-components";
import colors from "yoast-components/style-guide/colors.json";
import { addPlaceholderStyles } from "../styles/inputs";
import validate from "validate.js";
import { speak } from "@wordpress/a11y";
import _debounce from "lodash/debounce";
import ErrorDisplay from "../errors/ErrorDisplay";
import { ModalHeading } from "./Headings";
import ButtonsContainer from "./general/ButtonsContainer";
import YoastNativeSelect from "./general/YoastNativeSelect";
import { StyledLabel } from "./Labels";

const messages = defineMessages( {
	validationFormatURL: {
		id: "validation.format.url",
		defaultMessage: "Please enter a valid URL. Remember to use http:// or https://.",
	},
} );

const AddSiteModal = styled.div`
	max-width: 640px;
	margin: auto;
	font-size: 1em;
`;

const WebsiteURL = addPlaceholderStyles( styled.input`
	width: 100%;
	height: 48px;
	box-shadow: inset 0 2px 8px 0px rgba(0,0,0,0.3);
	background: ${ colors.$color_grey };
	padding: 0 0 0 10px;
	font-size: 1em;
	border: 0;
	margin-bottom: 8px;
` );

const ValidationText = styled.div`
	font-size: 1em;
	color: ${ colors.$color_red};
	margin: 1em 0;
	min-height: 1.8em;
`;

const WideLargeButton = makeButtonFullWidth( LargeButton );
const WideSecondaryButton = makeButtonFullWidth( LargeSecondaryButton );

/**
 * Returns the AddSite component.
 *
 * @returns {ReactElement} The AddSite component.
 */
class AddSite extends React.Component {
	/**
	 * Initializes the class with the specified props.
	 *
	 * @param {Object} props The props to be passed to the class that was extended from.
	 *
	 * @returns {void}
	 */
	constructor( props ) {
		super( props );

		this.constraints = {
			url: this.urlConstraints.bind( this ),
		};

		this.state = {
			validationError: null,
			showValidationError: false,
			selectedValue: "wordpress",
		};

		// Defines the debounced function to show the validation error.
		this.showValidationMessageDebounced  = _debounce( this.showValidationMessage, 1000 );
		// Defines the debounced function to announce the validation error.
		this.speakValidationMessageDebounced = _debounce( this.speakValidationMessage, 1000 );

		this.onWebsiteURLChange = this.onWebsiteURLChange.bind( this );
		this.handleOnSubmit     = this.handleOnSubmit.bind( this );
		this.handleOnBlur       = this.handleOnBlur.bind( this );
	}

	/**
	 * Tests whether customer has access to composer token feature. If so, returns a drop-down selector for CMS type.
	 * If not, returns null.
	 *
	 * @returns {*} Either null or html for the CMS type drop-down.
	 */
	getPlatformSelect() {
		return (
			<div>
				<StyledLabel
					htmlFor="add-site-select-platform"
				>
					<FormattedMessage
						id="sites.addSite.enterUrl"
						defaultMessage="Please select the platform that your website is running on:"
					/>
				</StyledLabel>
				<YoastNativeSelect
					selectId="add-site-select-platform"
					selectName="selectPlatform"
					selectDefaultValue={ this.state.selectedValue }
					selectOnBlur={ this.handleOnBlur }
				>
					<option value="wordpress">WordPress</option>
					<option value="typo3">TYPO3</option>
				</YoastNativeSelect>
			</div>
		);
	}

	/**
	 * Sets the constraints for validation to URL, and sets the message that should be returned if the constraints are not met.
	 *
	 * @returns {Object} Returns the constraints for the URL, and the message.
	 */
	urlConstraints() {
		return {
			url: {
				message: this.props.intl.formatMessage( messages.validationFormatURL ),
			},
		};
	}

	/**
	 * Calls onChange function when website url changes.
	 *
	 * @param {Object} event The event returned by the WebsiteURLChange.
	 *
	 * @returns {void}
	 */
	onWebsiteURLChange( event ) {
		const value = event.target.value;
		this.props.onChange( value );
	}

	/**
	 * Runs url validation and shows/hides error if validation returns error.
	 *
	 * @param {string}  url       The url to validate.
	 * @param {boolean} debounced Whether to show the debounced error message.
	 *
	 * @returns {void}
	 */
	runValidation( url, debounced = true ) {
		const validationError = this.validateUrl( url );
		if ( validationError ) {
			this.updateValidationMessage( validationError );
			if ( debounced ) {
				this.showValidationMessageDebounced();
				this.speakValidationMessageDebounced();
			} else {
				this.showValidationMessage();
				this.speakValidationMessage();
			}
		} else {
			this.updateValidationMessage( null );
			this.hideValidationError();
		}
	}

	/**
	 * Shows the validation error.
	 *
	 * @returns {void}
	 */
	showValidationMessage() {
		this.setState( { showValidationError: true } );
	}

	/**
	 * Updates the validation message.
	 *
	 * @param {string} message The validation message.
	 *
	 * @returns {void}
	 */
	updateValidationMessage( message ) {
		this.setState( {
			validationError: message,
		} );
	}

	/**
	 * Validates URL and shows validation error if URL is invalid.
	 *
	 * @param {string} input The URL to be validated.
	 * @returns {string} URL Validation error message.
	 */
	validateUrl( input = "" ) {
		if ( input === "" ) {
			return null;
		}

		const result = validate( { website: input }, { website: this.urlConstraints() }, { format: "detailed" } );

		if ( result && result[ 0 ] !== null ) {
			return result[ 0 ].options.message;
		}
		return null;
	}

	/**
	 * Hides the validation message and cancels a potential debounced error.
	 *
	 * @returns {void}
	 */
	hideValidationError() {
		this.setState( { showValidationError: false }, () => {
			this.showValidationMessageDebounced.cancel();
			this.speakValidationMessageDebounced.cancel();
		} );
	}

	/**
	 * Renders the validation error message container.
	 *
	 * When there are no errors, this container is rendered empty and acts like
	 * a "placeholder" taking space in the page to avoid "jumps" in the layout.
	 *
	 * @param {string} input The string in the input field.
	 * @returns {ReactElement} Returns a div that is either empty or contains an error message.
	 */
	urlValidityMessage( input = "" ) {
		if ( this.state.showValidationError && input !== "" ) {
			return (
				<ValidationText>
					<FormattedMessage
						id="sites.addSite.urlValidationMessage"
						defaultMessage={ this.state.validationError }
					/>
				</ValidationText>
			);
		}
		return (
			<ValidationText />
		);
	}

	/**
	 * Handles the submit event.
	 *
	 * @param {object} event The submit event.
	 *
	 * @returns {void}
	 */
	handleOnSubmit( event ) {
		event.preventDefault();
		if ( ! this.state.validationError && !! this.props.linkingSiteUrl ) {
			this.props.onConnectClick( this.state.selectedValue );
		}
	}

	/**
	 * Handles the blur event.
	 *
	 * @param {object} event The blur event.
	 *
	 * @returns {void}
	 */
	handleOnBlur( event ) {
		this.setState( {
			selectedValue: event.target.value,
		} );
	}

	/**
	 * Sends a message to the ARIA live assertive region.
	 *
	 * @returns {void}
	 */
	speakValidationMessage() {
		const message = this.props.intl.formatMessage( messages.validationFormatURL );
		speak( message, "assertive" );
	}

	/**
	 * Populates the URL input field with the search term, if any, and runs validation.
	 *
	 * @returns {void}
	 */
	componentDidMount() {
		if ( this.props.query ) {
			this.props.onChange( this.props.query );
			this.runValidation( this.props.query, false );
		}
	}

	/**
	 * Validates the URL when entering a value in the URL input field.
	 *
	 * @param {object} nextProps The new props the component will receive.
	 *
	 * @returns {void}
	 */
	componentWillReceiveProps( nextProps ) {
		if ( this.props.linkingSiteUrl !== nextProps.linkingSiteUrl ) {
			this.runValidation( nextProps.linkingSiteUrl );
		}
	}

	/**
	 * Cancels debounced functions when the component unmounts.
	 *
	 * @returns {void}
	 */
	componentWillUnmount() {
		this.showValidationMessageDebounced.cancel();
		this.speakValidationMessageDebounced.cancel();
	}

	/**
	 * Renders the component.
	 *
	 * @returns {ReactElement} The rendered component.
	 */
	render() {
		return (
			<AddSiteModal>
				<ModalHeading>
					<FormattedMessage id="sites.addSite.header" defaultMessage="Add Site" />
				</ModalHeading>

				<form onSubmit={ this.handleOnSubmit } noValidate={ true }>
					<StyledLabel htmlFor="add-site-input">
						<FormattedMessage
							id="sites.addSite.enterUrl"
							defaultMessage="Please enter the URL of the site you would like to add to your account:"
						/>
					</StyledLabel>

					<WebsiteURL
						type="url"
						id="add-site-input"
						placeholder={ "https://example-site.com" }
						value={ this.props.linkingSiteUrl }
						onChange={ this.onWebsiteURLChange }
					/>

					<ErrorDisplay error={ this.props.error } />

					{ this.getPlatformSelect() }
					{ this.urlValidityMessage( this.props.linkingSiteUrl ) }
					<ButtonsContainer>
						<WideSecondaryButton onClick={ this.props.onCancelClick }>
							<FormattedMessage id="sites.addSite.cancel" defaultMessage="Cancel" />
						</WideSecondaryButton>
						<WideLargeButton
							type="submit"
							enabledStyle={ ! this.state.validationError && !! this.props.linkingSiteUrl }
						>
							<FormattedMessage id="sites.addSite.connect" defaultMessage="Add" />
						</WideLargeButton>
					</ButtonsContainer>
				</form>
			</AddSiteModal>
		);
	}
}

AddSite.propTypes = {
	intl: intlShape.isRequired,
	linkingSiteUrl: PropTypes.string.isRequired,
	onCancelClick: PropTypes.func.isRequired,
	onConnectClick: PropTypes.func.isRequired,
	onChange: PropTypes.func.isRequired,
	query: PropTypes.string.isRequired,
	error: PropTypes.object,
};

AddSite.defaultProps = {
	error: {},
};

export default injectIntl( AddSite );
