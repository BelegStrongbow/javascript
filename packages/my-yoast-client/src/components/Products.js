import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";
import colors from "yoast-components/style-guide/colors.json";
import Product from "./Product";
import { ListHeading } from "./Headings";

const ProductsContainer = styled.div`
	display: flex;
	background-color: ${ colors.$color_white };
	padding: 0 24px 24px 24px;
	margin: 0 0 0 8px;
	flex-wrap: wrap;
	width: 100%;
	align-items: stretch;
`;

const ProductsListHeading = styled( ListHeading )`
	width: calc( 100% - 16px );
	font-size: 22px;
	margin: 0 0 24px 0px;
	padding: 0 0 24px 0;
`;

/**
 * Returns the rendered Products component.
 *
 * @param {Object} props The props to use.
 *
 * @returns {ReactElement} The rendered products page.
 */
export default class Products extends React.Component {
	/**
	 * Renders the component.
	 *
	 * @returns {ReactElement} The rendered component.
	 */
	render() {
		const props = this.props;
		return (
			<ProductsContainer>
				<ProductsListHeading>
					{ props.heading }
					{ props.byLine }
				</ProductsListHeading>
				{ props.noResults }
				{ props.products.map( function( product ) {
					return <Product
						key={ product.glNumber }
						name={ product.name }
						glNumber={ product.glNumber }
						currentVersion={ product.currentVersion }
						icon={ product.icon }
						buttons={ product.buttons }
						composerToken={ props.composerToken }
						onComposerHelpModalOpen={ props.onComposerHelpModalOpen }
						onComposerHelpModalClose={ props.onComposerHelpModalClose }
						type={ product.category }
					/>;
				} ) }
			</ProductsContainer>
		);
	}
}

Products.propTypes = {
	byLine: PropTypes.element,
	heading: PropTypes.string.isRequired,
	noResults: PropTypes.string,
	composerToken: PropTypes.object,
	onComposerHelpModalOpen: PropTypes.func,
	onComposerHelpModalClose: PropTypes.func,
	composerHelpProductName: PropTypes.string,
};

Products.defaultProps = {
	byLine: null,
	noResults: "",
	onComposerHelpModalClose: () => {},
	onComposerHelpModalOpen: () => {},
};
