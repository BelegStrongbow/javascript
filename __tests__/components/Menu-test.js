import React from 'react';
import renderer from 'react-test-renderer';

import { MainMenu, Subscriptions, Courses, Account, Sites } from '../../src/components/Menu';

test('the menu matches the snapshot', () => {
    const component = renderer.create(
        <MainMenu/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

//
// test('the button with button type given matches the snapshot', () => {
//     const component = renderer.create(
//         <Button type="button">ButtonValue</Button>
//     );
//
//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
// });
//
// test('the button handling an onclick event', () => {
//     const component = renderer.create(
//         <Button onClick={ () => {
//             return 'clicked';
//         } }>ButtonValue</Button>
//     );
//
//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
//
//     // manually trigger the callback
//     tree.props.onClick();
//
//     // re-rendering
//     tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
// });
