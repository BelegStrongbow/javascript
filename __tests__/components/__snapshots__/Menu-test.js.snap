// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the menu matches the snapshot 1`] = `
<nav
  className="bpAoaQ"
>
  <ul
    role="list"
  >
    <li>
      <a
        ariaCurrent="page"
        className="active-class-name esXMxt"
        href="/sites"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt=""
          className="fHJXFs"
          src="sites.svg"
        />
        Sites
      </a>
    </li>
    <li>
      <a
        ariaCurrent="page"
        className="esXMxt"
        href="/courses"
        onClick={[Function]}
        style={undefined}
      >
        <img
          alt=""
          className="fHJXFs"
          src="courses.svg"
        />
        Courses
      </a>
    </li>
    <li>
      <a
        ariaCurrent="page"
        className="esXMxt"
        href="/downloads"
        onClick={[Function]}
        style={undefined}
      >
        <img
          alt=""
          className="fHJXFs"
          src="download.svg"
        />
        Downloads
      </a>
    </li>
    <li>
      <a
        ariaCurrent="page"
        className="esXMxt"
        href="/account"
        onClick={[Function]}
        style={undefined}
      >
        <img
          alt=""
          className="fHJXFs"
          src="user.svg"
        />
        Account
      </a>
    </li>
  </ul>
</nav>
`;
